### PR TITLE
RTEMIS-121: Updated the static field.

### PR DESCRIPTION
**Pull request by jaswinsingh**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-121

---
### What the PR does
`(Explain fixes, changes and impacts)`
- Updated the static field.

### CODE REVIEW

**Review by kunal-sin — APPROVED** [review]

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php [hunk 1]
```diff
@@ -278,15 +371,26 @@ public function fetchSchoolList(&$form, FormStateInterface $form_state) {
...
+    if (!empty($selected_school_preference)) {
+      foreach ($selected_school_preference as $preferenceSchoolId) {
+        if (isset($available_schools[$preferenceSchoolId])) {
+          $sortedSchools[$preferenceSchoolId] = $available_schools[$preferenceSchoolId];
+          unset($available_schools[$preferenceSchoolId]);
+        }
       }
     }
-    return array_merge($sortedSchools, array_values($available_schools));
+    // Return array_merge($sortedSchools, array_values($available_schools));
```
Comment (kunal-sin) [inline modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php:392]: Can we remove the commented code?

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php [hunk 2]
```diff
@@ -278,15 +371,26 @@ public function fetchSchoolList(&$form, FormStateInterface $form_state) {
   /**
    * Sort the school preference based on entity id.
    */
-  private function sortPreferenceSchool($selected_school_preference, $available_schools) {
+  private function sortPreferenceSchool($selected_school_preference, $available_schools, $items) {
     $sortedSchools = [];
```
Comment (kunal-sin) [inline modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php:375]: `$sorted_schools`

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php [hunk 3]
```diff
@@ -70,27 +73,171 @@ public function buildForm(array $form, FormStateInterface $form_state) {
...
+      // Removed the `limit_validation_errors` attribute as it restrict
+      // values in form_state.
+      unset($form['field_siblings_details']['widget']['add_more']['add_more_button_siblings']['#limit_validation_errors']);
+      $form['field_has_siblings']['widget']['#ajax'] = $ajaxProperty;
+
+      // Fetch school list based on form_state values or on student edit form.
+      $availableSchools = [];
+      if ((isset($values['field_class'][0]['value']) && is_numeric($values['field_class'][0]['value'])) && isset($values['field_date_of_birth'][0]['value']) && $values['field_date_of_birth'][0]['value'] instanceof DrupalDateTime && !empty($values['field_gender'][0]['value']) && !empty($values['field_location'][0]['target_id']) || $this->getRouteMatch()->getRouteName() == 'entity.mini_node.edit_form') {
+        // Get the school list.
+        $availableSchools = $this->getSchoolPreferenceList($form, $form_state);
```
Comment (kunal-sin) [inline modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php:116]: `$available_schools`

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php (hunk 1 above)
Comment (jaswinsingh) [reply modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php]: Updated.

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php (hunk 2 above)
Comment (jaswinsingh) [reply modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php]: Updated.

File: modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php (hunk 3 above)
Comment (jaswinsingh) [reply modules/rte_mis_student/src/Form/OverrideMiniNodeForm.php]: Updated.

**Pull request merged**